### PR TITLE
Configurable connect and read timeouts

### PR DESCRIPTION
**Pull request by brandur-stripe**
Continues the work by @agori in #366 to add configurable connect and read timeouts on both the top-level `Stripe` class and on `RequestOptions` for finer grainer configurability.

### CODE REVIEW

**Comment by brandur-stripe** [comment]
Squashed @agori's changes in 3bede6e and patched mine on top in 0267646.

r? @ob-stripe Would you mind taking a look at this? Specifically the changes in commit 0267646 would be okay I think.

File: README.md [hunk 1]
```diff
@@ -94,3 +117,7 @@ unit and functional tests. For example:
     mvn test -D test=com.stripe.model.AccountTest
     mvn test -D test=com.stripe.functional.ChargeTest
     mvn test -D test=com.stripe.functional.ChargeTest#testChargeCreate
+
+<!--
+# vim: set tw=79:
+-->
```
Comment (brandur-stripe) [inline README.md]: I snuck another one of these in :$ IMO it's a good idea to try to get these in a state where they're easily readable as either plain text or rendered Markdown on GitHub, and consistent line length is a piece of that.

Comment (ob-stripe) [inline README.md]: 👍

**Comment by ob-stripe** [comment]
lgtm! Thanks @brandur-stripe

**Comment by brandur-stripe** [comment]
Thanks @ob-stripe!

**Comment by brandur-stripe** [comment]
Released as 4.5.0.

**Comment by agori** [comment]
Thank you both for having accepted and improved the patch! 
Much appreciated :)

**Comment by brandur-stripe** [comment]
@agori NP! Thanks for submitting!

**Pull request merged**